index.js

```diff
@@ -1115,17 +1124,23 @@ class Watcher extends ReadyResource {
   _closeSnapshots () {
```

Comment:
Somehow we could kind of use this same method in here:
https://github.com/holepunchto/hyperbee/blob/f3baf65dc494a63a34da97ca860fafc0a85bfa2c/index.js#L1070

Because it's doing something similar, maybe with a snapshot option added

Reply:
But I tried to keep the changes at minimal so the leak fix is clearer, but it could be a code reduction for later

Reply:
We can even remove the `closing` array, I don't remember why I added it but it doesn't seem to provide much value